src/main/resources/db/migration/V12__new_table_failed_groups.sql

```diff
@@ -0,0 +1,8 @@
+CREATE TABLE failed_groups (
+  campaign_id  CHAR(36) NOT NULL,
+  group_id     CHAR(36) NOT NULL,
+  failure_code VARCHAR(200) NOT NULL,
```

Comment:
Actually, it should be the PK because we'd like to have only one device group per camapign & failure code.

Reply:
Indeed, but as I see it the whole triplet is unique right? For a group and campaign there is one failure code, and for a group and failure code there is one campaign (unless we intend to try to reuse groups, which seems unreasonable). Hence the PK definition here.

Reply:
Having the whole triplet to be unique allows creating several groups for the same campaign and failure code because triplets `(cid, gid1, fc)` and `(cid, gid2, fc)` are different, and thus, unique. On the other hand, when we have only campaign ID and failure code as a PK, the triplets above become the same, preventing us from erroneously inserting another group for the same campaign and failure code.

Reply:
Damn, right! Fixed now.